bot/code_review_bot/workflow.py

```diff
@@ -215,6 +215,18 @@ def _build_tasks(tasks):
         # Publish issues in the backend
         self.backend_api.publish_issues(issues, revision)
 
+    def start_analysis(self, revision):
+        """
+        Apply a patch on a local clone and push to try to trigger a new Code review analysis
+        """
+        logger.info("Patch should be applied here...")
+
+        # TODO: clone upstream at tip
+
+        # TODO: apply stack of patches
```

Comment:
The current implementation doesn't revert the repository properly if a patch fails to apply. Example: A patch is being applied and adds file2 and modifies file1. There is a conflict for file1. If the patch is applied again by another review bot run, it will fail because file2 already exists. This had been reported last week by Julien. Let me know if I shall file a issue (the logs of this example are inaccessible now).

Reply:
This should probably be a followup, I'm focusing now on porting the current workflow from events+libmozevent into the bot, by reusing as much existing structure as possible.

This PR will rely on libmozevent dependency to avoid re-implementing the whole clone+apply stack + push workflow